PDF Viewer Annotations

silviyaboteva opened the pull request:
Closes: https://github.com/telerik/kendo-themes-private/issues/265

epetrow commented on tests/_output/fluent/dock-manager/dock-manager.png:
Is the change in the toolbar separator color here expected?

silviyaboteva replied:
yes: https://www.figma.com/design/hmTm63q1tFA73b4mVrKsSa/Spreadsheet?node-id=660-316870&node-type=frame&t=T1k3DJD0qi4f6yd9-0

epetrow reviewed (changes requested):
A few questions as well:

- The toolbars in Material look quite different now, I believe this is intentional?
- Toobarl separator in Fluent has also changed, again is it intentional?
- The position in the action buttons in the dialogs differs from the convention we have in other popups, editing popup for example. Have you discussed this with the design team?

silviyaboteva commented on packages/default/scss/pdf-viewer/_layout.scss:
```diff
@@ -85,9 +119,101 @@
+                    pointer-events: none;
+                    transform-origin: 0 0;
+
+                    .k-internal {
```
We are basing on the PDF.js to map their classes to create our kendo ones via adding the k- prefix. 
`internal` -> `k-internal`
Demo: https://mozilla.github.io/pdf.js/web/viewer.html

We are free to hear suggestions for renaming this element :)

silviyaboteva replied:
after an offlie sync with @epetrow, we decided to stick to this class for consistency with the pdf.js implementation

silviyaboteva commented:
Hey @epetrow,

> * The toolbars in Material look quite different now, I believe this is intentional? 

-> Which toolbars? the standalone omponent toolbar or the one that is part of the PDF-viewer. The standalone toolbar (solid by default) hasn't changed. In the scope of the PDF viewer, toolbar is in flat fillMode and this is synced with the design team and updated accordingly.

> * Toobarl separator in Fluent has also changed, again is it intentional? 

-> yes: https://www.figma.com/design/hmTm63q1tFA73b4mVrKsSa/Spreadsheet?node-id=660-316870&node-type=frame&t=T1k3DJD0qi4f6yd9-0

> * The position in the action buttons in the dialogs differs from the convention we have in other popups, editing popup for example. Have you discussed this with the design team? 

-> synced with @imasyuko and decided to reorder them in order the primary button to be the first one.

epetrow commented on packages/default/scss/pdf-viewer/_layout.scss:
```diff
@@ -33,61 +33,201 @@
+                        overflow: visible;
+                        white-space: nowrap;
+                        user-select: none;
+                        color: $kendo-pdf-viewer-free-text-annotation-text;
```
I've missed these guys ( here and line 224 ), but they should also go to `_theme.scss`

epetrow commented on packages/default/scss/pdf-viewer/_layout.scss:
```diff
@@ -33,61 +33,201 @@
-            flex: 1 1 auto;
-            outline: none;
+    // Canvas
+    .k-pdf-viewer-canvas {
```
Won't this change affect teams other than the pilot?

silviyaboteva replied:
I checked this. The rendering in all suites includes both classes at the same element `.k-canvas.k-pdf-viewer-canvas`. In the fluent theme we have used the `k-pdf-viewer-canvas` class for the CSS rules but in the default theme we have used the `k-canvas`. I made this change for consistency.

Here is the reference: https://github.com/telerik/kendo-themes/pull/5173/files/20a40e24f6584be88d494f81200ac2e9bbf63d35#diff-ecb5c4c15b8f8127171653f1208736404699fdaa31ce73207386069056b8558eL40

This said, there would be no affected team from this change :)

epetrow commented on packages/fluent/scss/pdf-viewer/_layout.scss:
```diff
@@ -78,16 +103,131 @@
+                        overflow: visible;
+                        white-space: nowrap;
+                        user-select: none;
+                        color: $kendo-pdf-viewer-free-text-annotation-text;
```
This too should be moved to `_theme.scss`

epetrow commented on packages/fluent/scss/pdf-viewer/_layout.scss:
```diff
@@ -78,16 +103,131 @@
+
+                    .k-internal:empty::before {
+                        content: attr(default-content);
+                        color: $kendo-pdf-viewer-free-text-annotation-placeholder-text;
```
Move to `_theme.scss`

epetrow reviewed (commented):
The commit messages look odd, maybe you've used squash instead of fixup, however we would not want the additional `chore`'s in the changelog.

Also the conditional rendering of the color editor action buttons seems more like a chore to me 🤔

epetrow commented on packages/fluent/scss/grid/_theme.scss:
Why are we moving these styles here?

silviyaboteva replied:
because in the Default theme they are not part of .k-column-menu. As part of .k-column-menu they do not allpy the border color and produces a black border. 
Also, if you notice the _layout.scss, the styles are separate from .k-column menu class as well.

epetrow commented on packages/default/scss/toolbar/_theme.scss:
The way that the fill mode styles for the toolbar are implemented differs from what we have as a sort of convention.

Currently the toolbar will not have a dedicated solid fill mode class since these styles are scoped to `k-toolbar`, which in my opinion is not ideal.

epetrow commented on packages/default/scss/pdf-viewer/_layout.scss:
```diff
@@ -33,61 +33,201 @@
-            flex: 1 1 auto;
-            outline: none;
+    // Canvas
+    .k-pdf-viewer-canvas {
```
All occurrences of `k-canvas` have been updated

inikolova commented on packages/fluent/scss/toolbar/_theme.scss:
```diff
@@ -22,6 +19,35 @@
+        );
+
+        > .k-separator {
+            border-inline-color: $kendo-toolbar-flat-border;
```
The correct syntax here should be: border-inline-color: var( --kendo-toolbar-flat-border, #{$kendo-toolbar-flat-border} );
